File: rust/src/models/subscription_activation_condition_enum.rs
// this file is @generated
use std::fmt;

use serde::{Deserialize, Serialize};

#[derive(
    Clone, Copy, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize,
)]
pub enum SubscriptionActivationConditionEnum {
    #[default]
    #[serde(rename = "ON_START")]
    OnStart,

    #[serde(rename = "ON_CHECKOUT")]
    OnCheckout,

    #[serde(rename = "MANUAL")]
    Manual,
}

impl fmt::Display for SubscriptionActivationConditionEnum {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let value = match self {
            Self::OnStart => "ON_START",
            Self::OnCheckout => "ON_CHECKOUT",
            Self::Manual => "MANUAL",
        };
        f.write_str(value)
    }
}

File: typescript/src/models/subscriptionActivationConditionEnum.ts
// this file is @generated

export enum SubscriptionActivationConditionEnum {
  OnStart = "ON_START",
  OnCheckout = "ON_CHECKOUT",
  Manual = "MANUAL",
}

export const SubscriptionActivationConditionEnumSerializer = {
  _fromJsonObject(object: any): SubscriptionActivationConditionEnum {
    return object;
  },

  _toJsonObject(self: SubscriptionActivationConditionEnum): any {
    return self;
  },
};
